### PR TITLE
Bug 1104764 - [Rooms][Settings] Delete rooms

### PR DESCRIPTION
**Pull request by crdlc**

### CODE REVIEW

File: app/js/screens/create_room.js [hunk 1]
```diff
@@ -138,7 +138,7 @@
     }).then((room) => {
       room.roomToken = token;
       Controller.onRoomCreated(room, userInteraction);
-      roomName === roomNameByDefault &&
+      (roomName === roomNameByDefault || !roomNameInput.value.trim().length) &&
```
Comment (crdlc) [inline app/js/screens/create_room.js]: Not related with this patch but I detected a bug where the counter was not incremented once the room was created

File: app/js/screens/delete_room.js [hunk 2]
```diff
@@ -4,16 +4,153 @@
...
+            return _showOfflineError();
+          } else {
+            return _deleteRemoteRooms(roomTokensAsOwner);
+          }
+        })
+
+        // 4º STEP: We have to remove rooms in DB once rooms have been removed
+        // remotely (all invited rooms + my rooms that were deleted from the
+        // server successfully)
+        .then(remoteRoomTokens => {
```
Comment (borjasalguero) [inline app/js/screens/delete_room.js]: s/remoteRoomTokens/remotelyRemovedTokens

File: app/js/screens/delete_room.js [hunk 3]
```diff
@@ -4,16 +4,153 @@
...
+          reject(error);
+        });
+      });
     });
   }
 
-  function _delete(token, isOwner) {
+  function _delete(params) {
+    var token = params.tokens[0];
+    var isOwner = params.ownership[0];
```
Comment (borjasalguero) [inline app/js/screens/delete_room.js]: I would use object approach istead of 2 arrays.

File: app/js/screens/calllog.js [hunk 4]
```diff
@@ -84,9 +84,12 @@
       name: 'Delete',
       l10nId: 'delete',
       method: function(room) {
-        LazyLoader.load('js/screens/delete_room.js', () => {
-          RoomDelete.show(room.roomToken,
-                          room.roomOwner === Controller.identity).catch(e => {
+        Loader.getRoomDelete().then(RoomDelete => {
+          var params = {
+            tokens: [room.roomToken],
+            ownership: [room.roomOwner === Controller.identity]
```
Comment (crdlc) [inline app/js/screens/calllog.js]: +1

File: app/js/screens/delete_room.js [hunk 5]
```diff
@@ -4,16 +4,153 @@
 
   var _ = navigator.mozL10n.get;
 
-  function _deleteRoom(token) {
+  var debug = Config.debug ? function() { console.log.apply(console, arguments) } :
```
Comment (crdlc) [inline app/js/screens/delete_room.js]: This is the first time that I add the debug flag, if you want I can get rid of him

File: app/js/screens/delete_room.js [hunk 6]
```diff
@@ -4,16 +4,153 @@
...
+  function _deleteAll() {
     return new Promise((resolve, reject) => {
-      Rooms.delete(token).then(() => {
-        Controller.onRoomDeleted(token);
-        resolve();
-      }, reject);
+      _showConfirmation('deleteAllRoomsConfirmation', function onDelete() {
+        var roomTokens;
+
+        // 1º STEP: Users are aware of deleting action is being performed
```
Comment (crdlc) [inline app/js/screens/delete_room.js]: ok

File: app/js/screens/delete_room.js (hunk 2 above)
Comment (crdlc) [inline app/js/screens/delete_room.js]: +1

File: app/js/screens/delete_room.js (hunk 3 above)
Comment (crdlc) [inline app/js/screens/delete_room.js]: ok!!

**Pull request merged**